### PR TITLE
Upgrade to Preact X

### PR DESCRIPTION
**Pull request by swashcap**
* **Problem:** Preact X changed [the `context` API](https://preactjs.com/guide/v10/context/), which this package uses to pass down `store`. A `connect`-ed component doesn't appear to render at all!
* **Solution:** Use the new `createContext` export and its consumer and provider.
* **Testing:** Just run `npm test`!

_Note:_ This PR should probably correspond to changing `peerDependencies.preact` to `>=10.x.x` in the _package.json_. I should be published to npm with a `next` tag or major version bump.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 
-export const Provider = function () {}
-Provider.prototype.getChildContext = function () {
-  return { store: this.props.store }
+export const StoreContext = createContext()
+
+export const Provider = (props) => {
+  const store = props.store
+  const rest = {}
+  Object.keys(props).forEach((key) => {
```
Comment (swashcap) [inline src/index.js:8]: microbundle doesn't like the spread operator. Just do it manually here.

File: src/index.js [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+    if (key !== 'store') {
+      rest[key] = props[key]
+    }
+  })
+
+  return h(StoreContext.Provider, Object.assign({ value: store }, rest))
```
Comment (swashcap) [inline src/index.js:14]: Map `props.store` to the context provider's expected prop, `value`.

File: src/index.js [hunk 3]
```diff
@@ -48,4 +56,8 @@ export const connect = (...args) => {
       return h(Comp, Object.assign({}, props, state, this.actionCreators))
     }
   }
+
+  return (props) => h(StoreContext.Consumer, undefined, (store) => {
```
Comment (swashcap) [inline src/index.js:60]: The only way to consume context in the new API is through the context `Consumer` or the [`useContext` hook](https://preactjs.com/guide/v10/hooks#usecontext). This seems like the simplest solution.

Comment (HenrikJoreteg) [reply src/index.js:60]: I like this, very clean.

**Pull request merged**